src/compiler/checker.ts

```diff
@@ -18355,6 +18388,57 @@ namespace ts {
             return forEach(symbol.declarations, d => isClassLike(d) ? d : undefined);
         }
 
+        function checkAugmentedPropertyMemberOverrides(type: InterfaceType, propertiesToCheck: Map<Symbol>): void {
+            // If the class does not explicitly declare 'extends Object',
+            // declarations that mask 'Object' members ('toString()', 'hasOwnProperty()', etc...)
+            // are considered here.
+
+            // check is disabled in ambient contexts
+            if (isInAmbientContext(type.symbol.valueDeclaration)) {
+                return;
+            }
+
+            const objectType = getSymbol(globals, "Object", SymbolFlags.Type);
+            if (!objectType) {
+                return;
+            }
+
+            for (const name in propertiesToCheck) {
+                const derived = getTargetSymbol(propertiesToCheck[name]);
+                const derivedDeclarationFlags = getDeclarationModifierFlagsFromSymbol(derived);
+                const found = objectType.members[name];
+                if (found) {
+                    if (compilerOptions.noImplicitOverride) {
+                        const foundSymbol = getTargetSymbol(found);
+                        let errorInfo = chainDiagnosticMessages(
+                            undefined,
+                            Diagnostics.Class_member_0_must_be_marked_override_when_noImplicitOverride_is_enabled_augmented_from_Object_1,
+                            symbolToString(derived),
+                            symbolToString(found)
+                        );
+                        if (!isPropertyIdenticalTo(derived, foundSymbol)) {

```

Comment:
This checks for identical properties, but the error message says "assignable to".

Reply:
Removed check (was unnecessary)